### PR TITLE
damage ability tests

### PR DESCRIPTION
**Pull request by Pentarctagon**

### CODE REVIEW

File: data/test/macros/wml_unit_test_macros.cfg [hunk 1]
```diff
@@ -116,29 +125,46 @@ bob#endarg
...
+            [/store_unit]
+
+            {ASSERT {VARIABLE_CONDITIONAL {ID1}.hitpoints numerical_equals "$(${ID1}.max_hitpoints-{DAMAGE1})"}}
+            {ASSERT {VARIABLE_CONDITIONAL {ID2}.hitpoints numerical_equals "$(${ID2}.max_hitpoints-{DAMAGE2})"}}
+
+            [modify_unit]
+                [filter]
+                    id={ID1},{ID2}
+                [/filter]
+                hitpoints = 1000
```
Comment (stevecotton) [inline data/test/macros/wml_unit_test_macros.cfg:161]: Doing this at the start of the loop would seem more logical to me, and that should make it easier to examine the state if the test fails too.

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/damage/damage_wfl_self.cfg [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+#####
+# API(s) being tested: [damage]value=
+##
+# Actions:
+# Give the leaders a damage ability that does a quarter its own max hp.
+# Have the side 1 leader attack the side 2 leader with both its weapons
+##
+# Expected end state:
+# The side 1 leader has 2 weapons both of which have 250 strikes (1000/4), self is the attacker
```
Comment (stevecotton) [inline data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/damage/damage_wfl_self.cfg:11]: "strikes" is an error here

File: data/test/macros/wml_unit_test_macros.cfg (hunk 1 above)
Comment (Pentarctagon) [reply data/test/macros/wml_unit_test_macros.cfg]: Updated.

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/damage/damage_wfl_self.cfg (hunk 2 above)
Comment (Pentarctagon) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/damage/damage_wfl_self.cfg]: Updated.

File: data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/damage/damage_wfl_self.cfg [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    [event]
+        name = start
+
+        [modify_unit]
+            [filter]
+            [/filter]
+            [effect]
+                apply_to = new_ability
+                [abilities]
+                    {TEST_ABILITY damage "(max_hitpoints/4)" () SELF=yes}
```
Comment (ProditorMagnus) [inline data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/damage/damage_wfl_self.cfg:23]: if both units have same max_hitpoints this does not test which unit is used

Comment (Pentarctagon) [reply data/test/scenarios/wml_tests/UnitsWML/AbilitiesWML/damage/damage_wfl_self.cfg:23]: Updated.

**Comment by Pentarctagon** [comment]
Will merge this in a couple days assuming no further comments.

**Pull request merged**